packages/react-components/src/components/chart/hooks/useTrendCursorsEvents.ts

```diff
@@ -55,50 +55,58 @@ const useTrendCursorsEvents = ({
     sizeRef.current = size;
     setGraphicRef.current = setGraphic;
     visualizationRef.current = visualization;
+    // disabling because graphics and series are stringified
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [seriesDep, size, isInCursorAddMode, setGraphic, isInSyncMode, graphicDep, visualization]);
 
   // shared add function between the context menu and on click action
-  const addNewTrendCursor = ({ posX, ignoreHotKey }: { posX: number; ignoreHotKey: boolean }) => {
-    // when adding through the context menu, we can ignore the hot key press
-    if ((ignoreHotKey || isInCursorAddModeRef.current) && graphicRef.current.length < MAX_TREND_CURSORS) {
+  const addNewTrendCursor = useCallback(
+    ({ posX, ignoreHotKey }: { posX: number; ignoreHotKey: boolean }) => {
+      // when adding through the context menu, we can ignore the hot key press
+      if ((ignoreHotKey || isInCursorAddModeRef.current) && graphicRef.current.length < MAX_TREND_CURSORS) {
+        if (isInSyncModeRef.current) {
+          const timestampInMs = calculateTimeStamp(posX, chartRef);
+          addTrendCursorsToSyncState({
+            groupId: groupId ?? '',
+            tcId: `trendCursor-${uuid()}`,
+            timestamp: timestampInMs,
+          });
+        } else {
+          const newTc = getNewTrendCursor({
+            size: sizeRef.current,
+            series: seriesRef.current,
+            x: posX,
+            chartRef,
+            visualization,
+          });
+
+          if (newTc) {
+            setGraphicRef.current([...graphicRef.current, newTc]);
+          }
+        }
+      }
+    },
+    [groupId, chartRef, sizeRef, seriesRef, visualization, addTrendCursorsToSyncState]
```

Comment:
the point of having refs like seriesRef was to not have this re run when the series changes, the ref will hold the latest value and can be referred directly